crates/opencascade/src/section.rs

```diff
@@ -0,0 +1,76 @@
+use crate::primitives::Shape;
+use cxx::UniquePtr;
+use opencascade_sys::ffi;
+
+/// A wrapper around the `BRepAlgoAPI_Section` class.
+pub struct Section {
+    pub(crate) inner: UniquePtr<ffi::BRepAlgoAPI_Section>,
+}
+impl Section {
+    /// Create a new `Section` to intersect `target` by `tool`.
+    pub fn new(target: &Shape, tool: &Shape) -> Section {
+        Section {
+            inner: ffi::BRepAlgoAPI_Section_ctor(
+                target.inner.as_ref().expect("Target was null"),
+                tool.inner.as_ref().expect("Tool was null"),
+            ),
+        }
+    }
+
+    /// Get the edges of the resulting intersection.
+    pub fn section_edges(self) -> Vec<Shape> {
+        // TODO: Given that the OCCT name is "SectionEdges", can we return a Vec<Edge>?
```

Comment:
Looking at the implementation code, it seems likely that would could cast these to `Edge`. But they went with a list of `Shape` instead so I'm not 100% sure it will always return `Edge`s.

Reply:
Agreed, I think it seems better to have the caller try that cast rather than create the possibility of a panic from an invalid cast.